internals-js/src/specs/connectSpec.ts

```diff
@@ -0,0 +1,183 @@
+import {DirectiveLocation, GraphQLError} from 'graphql';
+import { CorePurpose, FeatureDefinition, FeatureDefinitions, FeatureUrl, FeatureVersion } from "./coreSpec";
+import {
+  Schema,
+  NonNullType,
+  InputObjectType,
+  InputFieldDefinition,
+  ListType,
+} from '../definitions';
+import { registerKnownFeature } from '../knownCoreFeatures';
+import { createDirectiveSpecification } from '../directiveAndTypeSpecification';
+
+export const connectIdentity = 'https://specs.apollo.dev/connect';
+
+const CONNECT = "connect";
+const SOURCE = "source";
+const URL_PATH_TEMPLATE = "URLPathTemplate";
+const JSON_SELECTION = "JSONSelection";
+const CONNECT_HTTP = "ConnectHTTP";
+const SOURCE_HTTP = "SourceHTTP";
+const HTTP_HEADER_MAPPING = "HTTPHeaderMapping";
+
+export class ConnectSpecDefinition extends FeatureDefinition {
+  constructor(version: FeatureVersion, readonly minimumFederationVersion: FeatureVersion) {
+    super(new FeatureUrl(connectIdentity, CONNECT, version), minimumFederationVersion);
+
+    this.registerDirective(createDirectiveSpecification({
+      name: CONNECT,
+      locations: [DirectiveLocation.FIELD_DEFINITION],
+      repeatable: true,
+      // We "compose" these directives using the  `@join__directive` mechanism,
+      // so they do not need to be composed in the way passing `composes: true`
+      // here implies.
+      composes: false,
+    }));
+
+    this.registerDirective(createDirectiveSpecification({
+      name: SOURCE,
+      locations: [DirectiveLocation.SCHEMA],
+      repeatable: true,
+      composes: false,
+    }));
+
+    this.registerType({ name: URL_PATH_TEMPLATE, checkOrAdd: () => [] });
```

Comment:
I'm curious why we don't use `createScalarTypeSpecification()` here (a comment would be helpful if there's a reason, at least).

Reply:
using createScalarTypeSpecification — i noticed that there isn't `createInputObjectTypeSpecification`, is that something i should also add and use?

Reply:
https://github.com/apollographql/federation/pull/3118/commits/9f34ce4c6379554019ed156b67a49cf26b5aeb82